From 2da65f4fb4808389e2b71286426ce49f45fbae57 Mon Sep 17 00:00:00 2001
From: Paul Harrison <pharrison@mythtv.org>
Date: Mon, 24 Mar 2014 23:16:45 +0000
Subject: [PATCH 3/3] PlaylistEditorView: don't use itemVisible if using the
 tree view

This will make sure we don't hunt for node images and icons if we are using the
tree view which doesn't show them.

It would be nice to only disable the hunt if the buttonimage image is not
present in the buttonlist but I can't see a way to do that.
---
 mythplugins/mythmusic/mythmusic/playlisteditorview.cpp |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/mythplugins/mythmusic/mythmusic/playlisteditorview.cpp b/mythplugins/mythmusic/mythmusic/playlisteditorview.cpp
index 1d3ba49..c1bf27c 100644
--- a/mythplugins/mythmusic/mythmusic/playlisteditorview.cpp
+++ b/mythplugins/mythmusic/mythmusic/playlisteditorview.cpp
@@ -171,8 +171,10 @@ bool PlaylistEditorView::Create(void)
             this, SLOT(treeItemClicked(MythUIButtonListItem*)));
     connect(m_playlistTree, SIGNAL(nodeChanged(MythGenericTree*)),
             this, SLOT(treeNodeChanged(MythGenericTree*)));
-    connect(m_playlistTree, SIGNAL(itemVisible(MythUIButtonListItem*)),
-            this, SLOT(treeItemVisible(MythUIButtonListItem*)));
+
+    if (m_currentView == MV_PLAYLISTEDITORGALLERY)
+        connect(m_playlistTree, SIGNAL(itemVisible(MythUIButtonListItem*)),
+                this, SLOT(treeItemVisible(MythUIButtonListItem*)));
 
     BuildFocusList();
 
-- 
1.7.10.2

